File: home/home4.1/Pizza.swift
import Foundation

class Pizza {
    private var ingredients: [String]
    private var totalCost: Double

    init() {
        self.ingredients = []
        self.totalCost = 0.0
    }

    func addIngredient(_ ingredient: String, cost: Double) {
        ingredients.append(ingredient)
        totalCost += cost
    }

    func getIngredients() -> [String] {
        return ingredients
    }

    func getTotalCost() -> Double {
        return totalCost
    }
}

File: class/class4.1/pizza.cpp
#include "pizza.h"

void Pizza::add_ingredient(const std::string& ingredient, double cost) {
    ingredients.push_back(ingredient);
    total_cost += cost;
}

std::vector<std::string> Pizza::get_ingredients() const {
    return ingredients;
}

double Pizza::get_total_cost() const {
    return total_cost;
}
